mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
+
+    try:
+        ds = xarray.open_mfdataset(fileNames,
+                                   preprocess=preprocess_partial,
+                                   autoclose=autoclose, **kwargs)
+    except TypeError as e:
+        if 'autoclose' in str(e):
+            if autoclose:
```

Comment:
Why would we want to warn the user that autoclose is not supported if they don't want to use it?  This would mean that running with the current version of xarray spews warning after warning every time we read a data set.